services/patient.services.js

```diff
@@ -273,6 +273,100 @@ const editMedication = async (data, medicationId) => {
   }
 };
 
+const addPatientVital = async ({
+  patientId,
+  vitalId,
+  value,
+}) => {
+  const patient = await db.Patient.findOne({
+    where: { id: patientId },
+  });
+
+  if (patient === null) throw new NotFound("Patient not found.");
+
+  const vital = await db.Vital.findOne({
+    where: { id: vitalId },
+  });
+
+  if (vital === null) throw new NotFound("Vital not found.");
+
+  const optPatientVital = await db.PatientVital.findOne({
+    where: { patient_id: patientId, vital_id: vitalId, isArchived: false }
+  });
+
+  try {
+    if (optPatientVital !== null) {
```

Comment:
This method should in DB transaction. So if one action fails everything should be reverted in DB